pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -522,3 +804,237 @@ func prepareStoreForGarbageCollection(forTime time.Time, storeContainer string)
+		snapTime = snapTime.Add(time.Duration(time.Hour * 24 * 7))
+		if (mode == snapsInv1) || (mode == mixed && (i == 1 || i == 2)) {
+			dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+		} else {
+			dir = ""
+		}
+		snap := &brtypes.Snapshot{
+			Kind:          brtypes.SnapshotKindFull,
+			CreatedOn:     snapTime,
+			StartRevision: 0,
+			LastRevision:  1001,
+			SnapDir:       dir,
+		}
+		snap.GenerateSnapshotName()
+		expectedSnapList = append(expectedSnapList, snap)
+	}
+	fmt.Println("Weekly snapshot list prepared")
+	fmt.Printf("len: %d", len(expectedSnapList))
+
+	// daily snapshot
+	for i := 1; i <= 7; i++ {
+		snapTime = snapTime.Add(time.Duration(time.Hour * 24))
+		if mode == snapsInv1 {
+			dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+		} else {
+			dir = ""
+		}
+		snap := &brtypes.Snapshot{
+			Kind:          brtypes.SnapshotKindFull,
+			CreatedOn:     snapTime,
+			StartRevision: 0,
+			LastRevision:  1001,
+			SnapDir:       dir,
+		}
+		snap.GenerateSnapshotName()
+		expectedSnapList = append(expectedSnapList, snap)
+	}
+	fmt.Println("Daily snapshot list prepared")
+
+	// hourly snapshot
+	snapTime = snapTime.Add(time.Duration(time.Hour))
+	for now.Truncate(time.Hour).Sub(snapTime) > 0 {
+		if mode == snapsInv1 {
+			dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+		} else {
+			dir = ""
+		}
+		snap := &brtypes.Snapshot{
+			Kind:          brtypes.SnapshotKindFull,
+			CreatedOn:     snapTime,
+			StartRevision: 0,
+			LastRevision:  1001,
+			SnapDir:       dir,
+		}
+		snap.GenerateSnapshotName()
+		expectedSnapList = append(expectedSnapList, snap)
+		snapTime = snapTime.Add(time.Duration(time.Hour))
+	}
+	fmt.Println("Hourly snapshot list prepared")
+
+	// current hour
+	snapTime = now.Truncate(time.Hour)
+	if mode == snapsInv1 {
+		dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+	} else {
+		dir = ""
+	}
+	snap := &brtypes.Snapshot{
+		Kind:          brtypes.SnapshotKindFull,
+		CreatedOn:     snapTime,
+		StartRevision: 0,
+		LastRevision:  1001,
+		SnapDir:       dir,
+	}
+	snap.GenerateSnapshotName()
+	expectedSnapList = append(expectedSnapList, snap)
+	snapTime = snapTime.Add(time.Duration(time.Minute * 30))
+	for now.Sub(snapTime) >= 0 {
+		if mode == snapsInv1 {
+			dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+		} else {
+			dir = ""
+		}
+		snap := &brtypes.Snapshot{
+			Kind:          brtypes.SnapshotKindFull,
+			CreatedOn:     snapTime,
+			StartRevision: 0,
+			LastRevision:  1001,
+			SnapDir:       dir,
+		}
+		snap.GenerateSnapshotName()
+		expectedSnapList = append(expectedSnapList, snap)
+		snapTime = snapTime.Add(time.Duration(time.Minute * 30))
+	}
+	fmt.Println("Current hour full snapshot list prepared")
+
+	// delta snapshots
+	snapTime = snapTime.Add(time.Duration(-time.Minute * 30))
+	snapTime = snapTime.Add(time.Duration(time.Minute * 10))
+	for now.Sub(snapTime) >= 0 {
+		// if mode == "v1" {
+		// 	dir = fmt.Sprintf("Backup-%d", snapTime.Unix())
+		// } else {
+		// 	dir = ""
+		// }
```

Comment:
Please remove these lines if not required.

Reply:
Removed